sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_impl.hpp

```diff
@@ -107,33 +74,20 @@ double joint_matmul(TOperand *A, TOperand *B, TResult *C, queue &q, int i) {
           auto m1 = it.get_local_id(0);
           auto n1 = it.get_local_id(1) / sgSize;
           auto sg = it.get_sub_group();
-          joint_matrix<sub_group, TResult, use::accumulator, tM, tN>
-              tC[MCACHE1 / tM][NCACHE1 / tN]
+          joint_matrix<sub_group, TResult, use::accumulator, TM, TN>
+              tC[MCache1 / TM][NCache1 / TN]
 #ifdef INIT_LIST
-              = {joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>(),
-                 joint_matrix<sub_group, TResult, use::accumulator, tM, tN>()}
+              = {}; // default initialization of all array elements
```

Comment:
Could you check that LLVM IR which IGC gets is the same for INIT_LIST case before and after this change?
I have a feeling that it might be different and test is not verifying what it was...

Reply:
From what i see in the IR, we do not change low many times we load or store to the allocas. 

From looking at the implementation of the `joint_matrix` there is nothing to initialize (write to memory) in both cases (with `INIT_LIST` and without). The only payload for INTEL devices in `joint_matrix` is a pointer `spvm` that never gets set in the constructor. The only difference is the useless loops that get generated make sure the empty constructor gets run. After SPIRV reverse translation there is no constructor actually.

Reply:
great, thanks for checking